src/Sulu/Bundle/TagBundle/Admin/TagAdmin.php

```diff
@@ -23,6 +23,7 @@
 
 class TagAdmin extends Admin
 {
+    /** @var string */
```

Comment:
phpstan should understand the const and not error, so we should be possible to avoid this const `@var` phpdocs annotations

Reply:
Well kinda, when using it with self::PROPERTY then phpstan can figure it out easily. But when using static::PROPERTY which we're using in the definition it struggles and thinks the type of the property is int|string and explicitly typing it here fixes that.

So we could use self::PROPERTY everywhere but that might cause a BC break.

Reply:
There should in our classes `self::` or `SpecificAdmin::` used not `static::`.

Reply:
Also it would make sense to mark all `Admin` classes as final. As instead of extending just an own Admin instance should be created:

```php
/**
 * @final
 */
```

But that should be an own Pull Request. With using the PHPDoc it would not be a bc break.

Reply:
Yeah making the Admin classes final would be a great idea. But I will make a separate pull request.